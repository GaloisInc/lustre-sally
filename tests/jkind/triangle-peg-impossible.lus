-- Expected output:
--   VALID PROPERTIES: [prop]
-- K = 8

-- The 10-peg triangle peg board game:
--
--           1
--         2   3
--       4   5   6
--     7   8   9   10
--
-- This file shows that there is no solution

type peg = bool;

type loc = subrange [0, 10] of int;
const INVALID : loc = 0;

node peg(loc : loc; p1, p2, p3, p4, p5, p6, p7, p8, p9, p10 : peg)
returns (p : peg);
let
  p = if loc = 1 then p1 else
      if loc = 2 then p2 else
      if loc = 3 then p3 else
      if loc = 4 then p4 else
      if loc = 5 then p5 else
      if loc = 6 then p6 else
      if loc = 7 then p7 else
      if loc = 8 then p8 else
      if loc = 9 then p9 else
      if loc = 10 then p10 else
      false;
tel;

node mid(a, b : loc) returns (mid : loc);
let
  mid = if a = 1 and b = 4 then 2 else
        if a = 1 and b = 6 then 3 else
        if a = 2 and b = 7 then 4 else
        if a = 2 and b = 9 then 5 else
        if a = 3 and b = 8 then 5 else
        if a = 3 and b = 10 then 6 else
        if a = 4 and b = 1 then 2 else
        if a = 4 and b = 6 then 5 else
        if a = 6 and b = 1 then 3 else
        if a = 6 and b = 4 then 5 else
        if a = 7 and b = 2 then 4 else
        if a = 7 and b = 9 then 8 else
        if a = 8 and b = 3 then 5 else
        if a = 8 and b = 10 then 9 else
        if a = 9 and b = 2 then 5 else
        if a = 9 and b = 7 then 8 else
        if a = 10 and b = 3 then 6 else
        if a = 10 and b = 8 then 9 else
        INVALID;
tel;

node initial(p1, p2, p3, p4, p5, p6, p7, p8, p9, p10 : peg)
returns (initial : bool);
let
  initial = (not p1) and p2 and p3 and p4 and p5 and p6 and p7 and p8 and p9 and p10;
tel;

node b2i(x : bool) returns (y : int);
let
  y = if x then 1 else 0;
tel;

node solved(p1, p2, p3, p4, p5, p6, p7, p8, p9, p10 : peg)
returns (solved : bool);
let
  solved = (b2i(p1) + b2i(p2) + b2i(p3) + b2i(p4) + b2i(p5) +
            b2i(p6) + b2i(p7) + b2i(p8) + b2i(p9) + b2i(p10) = 1);
tel;

node unchanged_but(a, b, c : loc;
                   p1, p2, p3, p4, p5, p6, p7, p8, p9, p10 : peg)
returns (constraint : bool);
let
  constraint = true -> ((a = 1 or b = 1 or c = 1 or p1 = pre(p1)) and
                        (a = 2 or b = 2 or c = 2 or p2 = pre(p2)) and
                        (a = 3 or b = 3 or c = 3 or p3 = pre(p3)) and
                        (a = 4 or b = 4 or c = 4 or p4 = pre(p4)) and
                        (a = 5 or b = 5 or c = 5 or p5 = pre(p5)) and
                        (a = 6 or b = 6 or c = 6 or p6 = pre(p6)) and
                        (a = 7 or b = 7 or c = 7 or p7 = pre(p7)) and
                        (a = 8 or b = 8 or c = 8 or p8 = pre(p8)) and
                        (a = 9 or b = 9 or c = 9 or p9 = pre(p9)) and
                        (a = 10 or b = 10 or c = 10 or p10 = pre(p10)));
tel;

node historically(x : bool) returns (holds : bool);
let
  holds = x and (true -> pre holds);
tel;

node main(a, b : loc; p1, p2, p3, p4, p5, p6, p7, p8, p9, p10 : peg)
returns ();
var
    initial, valid_move, solved, prop : bool;
    mid : loc;
let
    initial = initial(p1, p2, p3, p4, p5, p6, p7, p8, p9, p10);
    solved = solved(p1, p2, p3, p4, p5, p6, p7, p8, p9, p10);

    mid = mid(a, b);
    valid_move = true ->
      (peg(a, pre p1, pre p2, pre p3, pre p4, pre p5,
              pre p6, pre p7, pre p8, pre p9, pre p10) and
       peg(mid, pre p1, pre p2, pre p3, pre p4, pre p5,
                pre p6, pre p7, pre p8, pre p9, pre p10) and
       not(peg(b, pre p1, pre p2, pre p3, pre p4, pre p5,
                  pre p6, pre p7, pre p8, pre p9, pre p10)) and
       not peg(a, p1, p2, p3, p4, p5, p6, p7, p8, p9, p10) and
       not peg(mid, p1, p2, p3, p4, p5, p6, p7, p8, p9, p10) and
       peg(b, p1, p2, p3, p4, p5, p6, p7, p8, p9, p10) and
       unchanged_but(a, b, mid, p1, p2, p3, p4, p5, p6, p7, p8, p9, p10));

    prop = not (historically(initial -> true) and historically(valid_move) and solved);
    --%PROPERTY prop;
tel;
