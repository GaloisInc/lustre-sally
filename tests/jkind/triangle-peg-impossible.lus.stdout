Sally Model
===========

(define-state-type
   S
   (
     (a Int)
     (b Int)
     (b2i Int)
     (c Int)
     (historically Bool)
     (initial Bool)
     (loc Int)
     (mid Int)
     (p1 Bool)
     (p10 Bool)
     (p2 Bool)
     (p3 Bool)
     (p4 Bool)
     (p5 Bool)
     (p6 Bool)
     (p7 Bool)
     (p8 Bool)
     (p9 Bool)
     (peg Bool)
     (prop Bool)
     (solved Bool)
     (x Bool)
     (|a:1| Int)
     (|a:2| Int)
     (|b2i:1| Int)
     (|b2i:2| Int)
     (|b2i:3| Int)
     (|b2i:4| Int)
     (|b2i:5| Int)
     (|b2i:6| Int)
     (|b2i:7| Int)
     (|b2i:8| Int)
     (|b2i:9| Int)
     (|b:1| Int)
     (|b:2| Int)
     (|gal-initializing| Bool)
     (|historically:1| Bool)
     (|init_pre_historically:1:init| Bool)
     (|init_pre_historically:1| Bool)
     (|init_pre_historically:init| Bool)
     (|init_pre_historically| Bool)
     (|loc:1| Int)
     (|loc:2| Int)
     (|loc:3| Int)
     (|loc:4| Int)
     (|loc:5| Int)
     (|p10:1| Bool)
     (|p10:2| Bool)
     (|p10:3| Bool)
     (|p10:4| Bool)
     (|p10:5| Bool)
     (|p10:6| Bool)
     (|p10:7| Bool)
     (|p10:8| Bool)
     (|p10:9| Bool)
     (|p1:1| Bool)
     (|p1:2| Bool)
     (|p1:3| Bool)
     (|p1:4| Bool)
     (|p1:5| Bool)
     (|p1:6| Bool)
     (|p1:7| Bool)
     (|p1:8| Bool)
     (|p1:9| Bool)
     (|p2:1| Bool)
     (|p2:2| Bool)
     (|p2:3| Bool)
     (|p2:4| Bool)
     (|p2:5| Bool)
     (|p2:6| Bool)
     (|p2:7| Bool)
     (|p2:8| Bool)
     (|p2:9| Bool)
     (|p3:1| Bool)
     (|p3:2| Bool)
     (|p3:3| Bool)
     (|p3:4| Bool)
     (|p3:5| Bool)
     (|p3:6| Bool)
     (|p3:7| Bool)
     (|p3:8| Bool)
     (|p3:9| Bool)
     (|p4:1| Bool)
     (|p4:2| Bool)
     (|p4:3| Bool)
     (|p4:4| Bool)
     (|p4:5| Bool)
     (|p4:6| Bool)
     (|p4:7| Bool)
     (|p4:8| Bool)
     (|p4:9| Bool)
     (|p5:1| Bool)
     (|p5:2| Bool)
     (|p5:3| Bool)
     (|p5:4| Bool)
     (|p5:5| Bool)
     (|p5:6| Bool)
     (|p5:7| Bool)
     (|p5:8| Bool)
     (|p5:9| Bool)
     (|p6:1| Bool)
     (|p6:2| Bool)
     (|p6:3| Bool)
     (|p6:4| Bool)
     (|p6:5| Bool)
     (|p6:6| Bool)
     (|p6:7| Bool)
     (|p6:8| Bool)
     (|p6:9| Bool)
     (|p7:1| Bool)
     (|p7:2| Bool)
     (|p7:3| Bool)
     (|p7:4| Bool)
     (|p7:5| Bool)
     (|p7:6| Bool)
     (|p7:7| Bool)
     (|p7:8| Bool)
     (|p7:9| Bool)
     (|p8:1| Bool)
     (|p8:2| Bool)
     (|p8:3| Bool)
     (|p8:4| Bool)
     (|p8:5| Bool)
     (|p8:6| Bool)
     (|p8:7| Bool)
     (|p8:8| Bool)
     (|p8:9| Bool)
     (|p9:1| Bool)
     (|p9:2| Bool)
     (|p9:3| Bool)
     (|p9:4| Bool)
     (|p9:5| Bool)
     (|p9:6| Bool)
     (|p9:7| Bool)
     (|p9:8| Bool)
     (|p9:9| Bool)
     (|peg:1| Bool)
     (|peg:2| Bool)
     (|peg:3| Bool)
     (|peg:4| Bool)
     (|peg:5| Bool)
     (|pre_historically:1| Bool)
     (|pre_historically| Bool)
     (|pre_p10| Bool)
     (|pre_p1| Bool)
     (|pre_p2| Bool)
     (|pre_p3| Bool)
     (|pre_p4| Bool)
     (|pre_p5| Bool)
     (|pre_p6| Bool)
     (|pre_p7| Bool)
     (|pre_p8| Bool)
     (|pre_p9| Bool)
     (|unchanged_but:init| Bool)
     (|unchanged_but| Bool)
     (|valid_move:init| Bool)
     (|valid_move| Bool)
     (|x:10| Bool)
     (|x:11| Bool)
     (|x:1:init| Bool)
     (|x:1| Bool)
     (|x:2| Bool)
     (|x:3| Bool)
     (|x:4| Bool)
     (|x:5| Bool)
     (|x:6| Bool)
     (|x:7| Bool)
     (|x:8| Bool)
     (|x:9| Bool))
   (
     (|a:2| Int)
     (|b:2| Int)
     (|p10:9| Bool)
     (|p1:9| Bool)
     (|p2:9| Bool)
     (|p3:9| Bool)
     (|p4:9| Bool)
     (|p5:9| Bool)
     (|p6:9| Bool)
     (|p7:9| Bool)
     (|p8:9| Bool)
     (|p9:9| Bool)))
(define-transition-system
   TS
   S
   (and
      (= |gal-initializing| true)
      (= |unchanged_but:init| true)
      (= |valid_move:init| true)
      (= |x:1:init| true)
      (= |init_pre_historically:1:init| true)
      (= |init_pre_historically:init| true))
   (and
      (= next.|gal-initializing| false)
      (= next.|a:2| input.|a:2|)
      (= next.|b:2| input.|b:2|)
      (= next.|p1:9| input.|p1:9|)
      (= next.|p2:9| input.|p2:9|)
      (= next.|p3:9| input.|p3:9|)
      (= next.|p4:9| input.|p4:9|)
      (= next.|p5:9| input.|p5:9|)
      (= next.|p6:9| input.|p6:9|)
      (= next.|p7:9| input.|p7:9|)
      (= next.|p8:9| input.|p8:9|)
      (= next.|p9:9| input.|p9:9|)
      (= next.|p10:9| input.|p10:9|)
      (= next.|p1:8| next.|p1:9|)
      (= next.|p2:8| next.|p2:9|)
      (= next.|p3:8| next.|p3:9|)
      (= next.|p4:8| next.|p4:9|)
      (= next.|p5:8| next.|p5:9|)
      (= next.|p6:8| next.|p6:9|)
      (= next.|p7:8| next.|p7:9|)
      (= next.|p8:8| next.|p8:9|)
      (= next.|p9:8| next.|p9:9|)
      (= next.|p10:8| next.|p10:9|)
      (=
         next.initial
         (and
            (not next.|p1:8|)
            next.|p2:8|
            next.|p3:8|
            next.|p4:8|
            next.|p5:8|
            next.|p6:8|
            next.|p7:8|
            next.|p8:8|
            next.|p9:8|
            next.|p10:8|))
      (= next.|loc:5| next.|a:2|)
      (= next.|p1:7| state.|p1:9|)
      (= next.|p2:7| state.|p2:9|)
      (= next.|p3:7| state.|p3:9|)
      (= next.|p4:7| state.|p4:9|)
      (= next.|p5:7| state.|p5:9|)
      (= next.|p6:7| state.|p6:9|)
      (= next.|p7:7| state.|p7:9|)
      (= next.|p8:7| state.|p8:9|)
      (= next.|p9:7| state.|p9:9|)
      (= next.|p10:7| state.|p10:9|)
      (=
         next.|peg:5|
         (ite
            (= next.|loc:5| 1)
            next.|p1:7|
            (ite
               (= next.|loc:5| 2)
               next.|p2:7|
               (ite
                  (= next.|loc:5| 3)
                  next.|p3:7|
                  (ite
                     (= next.|loc:5| 4)
                     next.|p4:7|
                     (ite
                        (= next.|loc:5| 5)
                        next.|p5:7|
                        (ite
                           (= next.|loc:5| 6)
                           next.|p6:7|
                           (ite
                              (= next.|loc:5| 7)
                              next.|p7:7|
                              (ite
                                 (= next.|loc:5| 8)
                                 next.|p8:7|
                                 (ite
                                    (= next.|loc:5| 9)
                                    next.|p9:7|
                                    (ite
                                       (= next.|loc:5| 10)
                                       next.|p10:7|
                                       false)))))))))))
      (= next.|a:1| next.|a:2|)
      (= next.|b:1| next.|b:2|)
      (=
         next.mid
         (ite
            (and
               (= next.|a:1| 1)
               (= next.|b:1| 4))
            2
            (ite
               (and
                  (= next.|a:1| 1)
                  (= next.|b:1| 6))
               3
               (ite
                  (and
                     (= next.|a:1| 2)
                     (= next.|b:1| 7))
                  4
                  (ite
                     (and
                        (= next.|a:1| 2)
                        (= next.|b:1| 9))
                     5
                     (ite
                        (and
                           (= next.|a:1| 3)
                           (= next.|b:1| 8))
                        5
                        (ite
                           (and
                              (= next.|a:1| 3)
                              (= next.|b:1| 10))
                           6
                           (ite
                              (and
                                 (= next.|a:1| 4)
                                 (= next.|b:1| 1))
                              2
                              (ite
                                 (and
                                    (= next.|a:1| 4)
                                    (= next.|b:1| 6))
                                 5
                                 (ite
                                    (and
                                       (= next.|a:1| 6)
                                       (= next.|b:1| 1))
                                    3
                                    (ite
                                       (and
                                          (= next.|a:1| 6)
                                          (= next.|b:1| 4))
                                       5
                                       (ite
                                          (and
                                             (= next.|a:1| 7)
                                             (= next.|b:1| 2))
                                          4
                                          (ite
                                             (and
                                                (= next.|a:1| 7)
                                                (= next.|b:1| 9))
                                             8
                                             (ite
                                                (and
                                                   (= next.|a:1| 8)
                                                   (= next.|b:1| 3))
                                                5
                                                (ite
                                                   (and
                                                      (= next.|a:1| 8)
                                                      (= next.|b:1| 10))
                                                   9
                                                   (ite
                                                      (and
                                                         (= next.|a:1| 9)
                                                         (= next.|b:1| 2))
                                                      5
                                                      (ite
                                                         (and
                                                            (= next.|a:1| 9)
                                                            (= next.|b:1| 7))
                                                         8
                                                         (ite
                                                            (and
                                                               (= next.|a:1| 10)
                                                               (= next.|b:1| 3))
                                                            6
                                                            (ite
                                                               (and
                                                                  (= next.|a:1| 10)
                                                                  (= next.|b:1| 8))
                                                               9
                                                               0)))))))))))))))))))
      (= next.|loc:4| next.mid)
      (= next.|p1:6| state.|p1:9|)
      (= next.|p2:6| state.|p2:9|)
      (= next.|p3:6| state.|p3:9|)
      (= next.|p4:6| state.|p4:9|)
      (= next.|p5:6| state.|p5:9|)
      (= next.|p6:6| state.|p6:9|)
      (= next.|p7:6| state.|p7:9|)
      (= next.|p8:6| state.|p8:9|)
      (= next.|p9:6| state.|p9:9|)
      (= next.|p10:6| state.|p10:9|)
      (=
         next.|peg:4|
         (ite
            (= next.|loc:4| 1)
            next.|p1:6|
            (ite
               (= next.|loc:4| 2)
               next.|p2:6|
               (ite
                  (= next.|loc:4| 3)
                  next.|p3:6|
                  (ite
                     (= next.|loc:4| 4)
                     next.|p4:6|
                     (ite
                        (= next.|loc:4| 5)
                        next.|p5:6|
                        (ite
                           (= next.|loc:4| 6)
                           next.|p6:6|
                           (ite
                              (= next.|loc:4| 7)
                              next.|p7:6|
                              (ite
                                 (= next.|loc:4| 8)
                                 next.|p8:6|
                                 (ite
                                    (= next.|loc:4| 9)
                                    next.|p9:6|
                                    (ite
                                       (= next.|loc:4| 10)
                                       next.|p10:6|
                                       false)))))))))))
      (= next.|loc:3| next.|b:2|)
      (= next.|p1:5| state.|p1:9|)
      (= next.|p2:5| state.|p2:9|)
      (= next.|p3:5| state.|p3:9|)
      (= next.|p4:5| state.|p4:9|)
      (= next.|p5:5| state.|p5:9|)
      (= next.|p6:5| state.|p6:9|)
      (= next.|p7:5| state.|p7:9|)
      (= next.|p8:5| state.|p8:9|)
      (= next.|p9:5| state.|p9:9|)
      (= next.|p10:5| state.|p10:9|)
      (=
         next.|peg:3|
         (ite
            (= next.|loc:3| 1)
            next.|p1:5|
            (ite
               (= next.|loc:3| 2)
               next.|p2:5|
               (ite
                  (= next.|loc:3| 3)
                  next.|p3:5|
                  (ite
                     (= next.|loc:3| 4)
                     next.|p4:5|
                     (ite
                        (= next.|loc:3| 5)
                        next.|p5:5|
                        (ite
                           (= next.|loc:3| 6)
                           next.|p6:5|
                           (ite
                              (= next.|loc:3| 7)
                              next.|p7:5|
                              (ite
                                 (= next.|loc:3| 8)
                                 next.|p8:5|
                                 (ite
                                    (= next.|loc:3| 9)
                                    next.|p9:5|
                                    (ite
                                       (= next.|loc:3| 10)
                                       next.|p10:5|
                                       false)))))))))))
      (= next.|loc:2| next.|a:2|)
      (= next.|p1:4| next.|p1:9|)
      (= next.|p2:4| next.|p2:9|)
      (= next.|p3:4| next.|p3:9|)
      (= next.|p4:4| next.|p4:9|)
      (= next.|p5:4| next.|p5:9|)
      (= next.|p6:4| next.|p6:9|)
      (= next.|p7:4| next.|p7:9|)
      (= next.|p8:4| next.|p8:9|)
      (= next.|p9:4| next.|p9:9|)
      (= next.|p10:4| next.|p10:9|)
      (=
         next.|peg:2|
         (ite
            (= next.|loc:2| 1)
            next.|p1:4|
            (ite
               (= next.|loc:2| 2)
               next.|p2:4|
               (ite
                  (= next.|loc:2| 3)
                  next.|p3:4|
                  (ite
                     (= next.|loc:2| 4)
                     next.|p4:4|
                     (ite
                        (= next.|loc:2| 5)
                        next.|p5:4|
                        (ite
                           (= next.|loc:2| 6)
                           next.|p6:4|
                           (ite
                              (= next.|loc:2| 7)
                              next.|p7:4|
                              (ite
                                 (= next.|loc:2| 8)
                                 next.|p8:4|
                                 (ite
                                    (= next.|loc:2| 9)
                                    next.|p9:4|
                                    (ite
                                       (= next.|loc:2| 10)
                                       next.|p10:4|
                                       false)))))))))))
      (= next.|loc:1| next.mid)
      (= next.|p1:3| next.|p1:9|)
      (= next.|p2:3| next.|p2:9|)
      (= next.|p3:3| next.|p3:9|)
      (= next.|p4:3| next.|p4:9|)
      (= next.|p5:3| next.|p5:9|)
      (= next.|p6:3| next.|p6:9|)
      (= next.|p7:3| next.|p7:9|)
      (= next.|p8:3| next.|p8:9|)
      (= next.|p9:3| next.|p9:9|)
      (= next.|p10:3| next.|p10:9|)
      (=
         next.|peg:1|
         (ite
            (= next.|loc:1| 1)
            next.|p1:3|
            (ite
               (= next.|loc:1| 2)
               next.|p2:3|
               (ite
                  (= next.|loc:1| 3)
                  next.|p3:3|
                  (ite
                     (= next.|loc:1| 4)
                     next.|p4:3|
                     (ite
                        (= next.|loc:1| 5)
                        next.|p5:3|
                        (ite
                           (= next.|loc:1| 6)
                           next.|p6:3|
                           (ite
                              (= next.|loc:1| 7)
                              next.|p7:3|
                              (ite
                                 (= next.|loc:1| 8)
                                 next.|p8:3|
                                 (ite
                                    (= next.|loc:1| 9)
                                    next.|p9:3|
                                    (ite
                                       (= next.|loc:1| 10)
                                       next.|p10:3|
                                       false)))))))))))
      (= next.loc next.|b:2|)
      (= next.|p1:2| next.|p1:9|)
      (= next.|p2:2| next.|p2:9|)
      (= next.|p3:2| next.|p3:9|)
      (= next.|p4:2| next.|p4:9|)
      (= next.|p5:2| next.|p5:9|)
      (= next.|p6:2| next.|p6:9|)
      (= next.|p7:2| next.|p7:9|)
      (= next.|p8:2| next.|p8:9|)
      (= next.|p9:2| next.|p9:9|)
      (= next.|p10:2| next.|p10:9|)
      (=
         next.peg
         (ite
            (= next.loc 1)
            next.|p1:2|
            (ite
               (= next.loc 2)
               next.|p2:2|
               (ite
                  (= next.loc 3)
                  next.|p3:2|
                  (ite
                     (= next.loc 4)
                     next.|p4:2|
                     (ite
                        (= next.loc 5)
                        next.|p5:2|
                        (ite
                           (= next.loc 6)
                           next.|p6:2|
                           (ite
                              (= next.loc 7)
                              next.|p7:2|
                              (ite
                                 (= next.loc 8)
                                 next.|p8:2|
                                 (ite
                                    (= next.loc 9)
                                    next.|p9:2|
                                    (ite
                                       (= next.loc 10)
                                       next.|p10:2|
                                       false)))))))))))
      (= next.a next.|a:2|)
      (= next.b next.|b:2|)
      (= next.c next.mid)
      (= next.|p1:1| next.|p1:9|)
      (= next.|p2:1| next.|p2:9|)
      (= next.|p3:1| next.|p3:9|)
      (= next.|p4:1| next.|p4:9|)
      (= next.|p5:1| next.|p5:9|)
      (= next.|p6:1| next.|p6:9|)
      (= next.|p7:1| next.|p7:9|)
      (= next.|p8:1| next.|p8:9|)
      (= next.|p9:1| next.|p9:9|)
      (= next.|p10:1| next.|p10:9|)
      (= next.|pre_p1| state.|p1:1|)
      (= next.|pre_p2| state.|p2:1|)
      (= next.|pre_p3| state.|p3:1|)
      (= next.|pre_p4| state.|p4:1|)
      (= next.|pre_p5| state.|p5:1|)
      (= next.|pre_p6| state.|p6:1|)
      (= next.|pre_p7| state.|p7:1|)
      (= next.|pre_p8| state.|p8:1|)
      (= next.|pre_p9| state.|p9:1|)
      (= next.|pre_p10| state.|p10:1|)
      (ite
         state.|unchanged_but:init|
         (= next.|unchanged_but| true)
         (=
            next.|unchanged_but|
            (and
               (or
                  (= next.a 1)
                  (= next.b 1)
                  (= next.c 1)
                  (= next.|p1:1| next.|pre_p1|))
               (or
                  (= next.a 2)
                  (= next.b 2)
                  (= next.c 2)
                  (= next.|p2:1| next.|pre_p2|))
               (or
                  (= next.a 3)
                  (= next.b 3)
                  (= next.c 3)
                  (= next.|p3:1| next.|pre_p3|))
               (or
                  (= next.a 4)
                  (= next.b 4)
                  (= next.c 4)
                  (= next.|p4:1| next.|pre_p4|))
               (or
                  (= next.a 5)
                  (= next.b 5)
                  (= next.c 5)
                  (= next.|p5:1| next.|pre_p5|))
               (or
                  (= next.a 6)
                  (= next.b 6)
                  (= next.c 6)
                  (= next.|p6:1| next.|pre_p6|))
               (or
                  (= next.a 7)
                  (= next.b 7)
                  (= next.c 7)
                  (= next.|p7:1| next.|pre_p7|))
               (or
                  (= next.a 8)
                  (= next.b 8)
                  (= next.c 8)
                  (= next.|p8:1| next.|pre_p8|))
               (or
                  (= next.a 9)
                  (= next.b 9)
                  (= next.c 9)
                  (= next.|p9:1| next.|pre_p9|))
               (or
                  (= next.a 10)
                  (= next.b 10)
                  (= next.c 10)
                  (= next.|p10:1| next.|pre_p10|)))))
      (= next.|unchanged_but:init| false)
      (ite
         state.|valid_move:init|
         (= next.|valid_move| true)
         (=
            next.|valid_move|
            (and
               next.|peg:5|
               next.|peg:4|
               (not next.|peg:3|)
               (not next.|peg:2|)
               (not next.|peg:1|)
               next.peg
               next.|unchanged_but|)))
      (= next.|valid_move:init| false)
      (= next.p10 next.|p10:9|)
      (= next.|x:11| next.p10)
      (=
         next.|b2i:9|
         (ite next.|x:11| 1 0))
      (= next.p9 next.|p9:9|)
      (= next.|x:10| next.p9)
      (=
         next.|b2i:8|
         (ite next.|x:10| 1 0))
      (= next.p8 next.|p8:9|)
      (= next.|x:9| next.p8)
      (=
         next.|b2i:7|
         (ite next.|x:9| 1 0))
      (= next.p7 next.|p7:9|)
      (= next.|x:8| next.p7)
      (=
         next.|b2i:6|
         (ite next.|x:8| 1 0))
      (= next.p6 next.|p6:9|)
      (= next.|x:7| next.p6)
      (=
         next.|b2i:5|
         (ite next.|x:7| 1 0))
      (= next.p5 next.|p5:9|)
      (= next.|x:6| next.p5)
      (=
         next.|b2i:4|
         (ite next.|x:6| 1 0))
      (= next.p4 next.|p4:9|)
      (= next.|x:5| next.p4)
      (=
         next.|b2i:3|
         (ite next.|x:5| 1 0))
      (= next.p3 next.|p3:9|)
      (= next.|x:4| next.p3)
      (=
         next.|b2i:2|
         (ite next.|x:4| 1 0))
      (= next.p2 next.|p2:9|)
      (= next.|x:3| next.p2)
      (=
         next.|b2i:1|
         (ite next.|x:3| 1 0))
      (= next.p1 next.|p1:9|)
      (= next.|x:2| next.p1)
      (=
         next.b2i
         (ite next.|x:2| 1 0))
      (=
         next.solved
         (=
            (+
               (+
                  (+
                     (+
                        (+
                           (+
                              (+
                                 (+
                                    (+ next.b2i next.|b2i:1|)
                                    next.|b2i:2|)
                                 next.|b2i:3|)
                              next.|b2i:4|)
                           next.|b2i:5|)
                        next.|b2i:6|)
                     next.|b2i:7|)
                  next.|b2i:8|)
               next.|b2i:9|)
            1))
      (ite
         state.|x:1:init|
         (= next.|x:1| next.initial)
         (= next.|x:1| true))
      (= next.|x:1:init| false)
      (= next.|pre_historically:1| state.|historically:1|)
      (ite
         state.|init_pre_historically:1:init|
         (= next.|init_pre_historically:1| true)
         (= next.|init_pre_historically:1| next.|pre_historically:1|))
      (= next.|init_pre_historically:1:init| false)
      (=
         next.|historically:1|
         (and next.|x:1| next.|init_pre_historically:1|))
      (= next.x next.|valid_move|)
      (= next.|pre_historically| state.historically)
      (ite
         state.|init_pre_historically:init|
         (= next.|init_pre_historically| true)
         (= next.|init_pre_historically| next.|pre_historically|))
      (= next.|init_pre_historically:init| false)
      (=
         next.historically
         (and next.x next.|init_pre_historically|))
      (=
         next.prop
         (not
            (and next.|historically:1| next.historically next.solved)))))

(query
   TS
   (or |gal-initializing| prop))

[Lustre] Validating properties:
[Lustre] Property prop...
  [Sally][Valid] using 9 past states
[Lustre] Model status: [Valid]
